app/ui-react/syndesis/cypress/integration/create-integration.spec.js

```diff
@@ -0,0 +1,281 @@
+describe('Create an Integration', () => {
+  const randomInteger = () => {
+    return Math.floor(Math.random() * (100 + 1));
+  };
+  const nameInt = randomInteger();
+  const connectionName = 'E2E Todo Connection ' + nameInt;
+  const integrationName = 'E2E Todo Integration ' + nameInt;
+  const connectionSlug = 'e2e-todo-connection-' + nameInt;
+  const integrationSlug = 'e2e-todo-integration-' + nameInt;
+
+  function createConnection() {
+    /**
+     * Create To Do Connection
+     * NOTE: This will break if a Connection with the same name already exists.
+     */
+    cy.visit('/');
+
+    cy.get('[data-testid=dashboard-create-connection-button]').click();
+
+    cy.location('pathname').should(
+      'contain',
+      '/connections/create/connection-basics'
+    );
+
+    cy.get('[data-testid=connection-card-todo-app-api-card]').click();
+
+    cy.location('pathname').should('contain', '/configure-fields');
+
+    cy.get('[data-testid=username]')
+      .clear()
+      .type(Cypress.env('connectorTodoUser'))
+      .should('have.value', Cypress.env('connectorTodoUser'));
+
+    cy.get('[data-testid=password]')
+      .clear()
+      .type(Cypress.env('connectorTodoPassword'))
+      .should('have.value', Cypress.env('connectorTodoPassword'));
+
+    cy.get('[data-testid=connection-creator-layout-next-button]')
+      .should('not.be.disabled')
+      .click();
+
+    cy.get('[data-testid=name]')
+      .clear()
+      .type(connectionName);
+
+    cy.get('[data-testid=description]')
+      .clear()
+      .type('Subscribe for and publish messages.');
+
+    cy.get('[data-testid=connection-creator-layout-next-button]').click();
+
+    cy.location('pathname').should('eq', '/connections/');
+
+    cy.get('.form-control').type(connectionName + '{enter}');
+    cy.get('[data-testid|=connection-card-' + connectionSlug + ']').should(
+      'exist'
+    );
+  }
+
+  function deleteConnection() {
```

Comment:
I dunno, this is duplicated from the previous file.  If the initial state needs to be a certain way there's a "test mode" that can be enabled for the backend that adds operations to truncate the database at will, in which case these tests could pre-load whatever state is needed that's outside of doing the actual test.

Reply:
This flag -> https://github.com/syndesisio/syndesis/blob/master/tools/bin/commands/install#L34

I think QE uses it for example.

Reply:
Cool, I'll look into that. 👍